app/views/landing-pages/league/PageLeagueGlobal.vue

```diff
@@ -44,13 +44,15 @@ export default {
     this.clanIdOrSlug = this.$route.params.idOrSlug || null
     this.findIdOfParam()
     // Would be odd to arrive here with ?registering=true and be logged out...
-    this.leagueSignupModalOpen = this.canRegister() && !!this.$route.query.registering
+    this.doneRegistering = !!this.$route.query.registered
```

Comment:
Instead of juggling the complex state and state management on the page, I made it a clean experience for newly signed up users. If you come back to /league and have registered, you'll still see `Join Now` but that's an ok compromise for the moment.